test/nodes/test_retriever.py

```diff
@@ -650,6 +650,37 @@ def test_table_text_retriever_training(tmp_path, document_store, samples_path):
     )
 
 
+def test_sentence_transformers_retriever_training_with_gradient_checkpointing():
+    with patch("sentence_transformers.SentenceTransformer.fit") as mock_fit:
+        retriever = EmbeddingRetriever(
+            embedding_model="sentence-transformers/all-MiniLM-L6-v2",
+            model_format="sentence_transformers",
+            use_gpu=False,
+        )
```

Comment:
I'm really sorry I didn't notice it earlier :see_no_evil: but is this initialization loading the model? If yes, can we mock the model as well so that we don't actually download and load the weights from hf?

If we can do that we'll be able to add the `@pytest.mark.unit` marker to this test. Otherwise it won't run in CI.